state-sync/state-sync-v1/src/network.rs

```diff
@@ -56,10 +60,32 @@ impl NewNetworkSender for StateSyncSender {
     }
 }
 
-impl StateSyncSender {
-    pub fn send_to(&mut self, recipient: PeerId, message: StateSyncMessage) -> Result<(), Error> {
+#[async_trait]
+impl ApplicationNetworkSender<StateSyncMessage> for StateSyncSender {
+    fn send_to(
+        &mut self,
+        recipient: PeerId,
+        message: StateSyncMessage,
+    ) -> Result<(), NetworkError> {
         let protocol = ProtocolId::StateSyncDirectSend;
-        Ok(self.inner.send_to(recipient, protocol, message)?)
+        self.inner.send_to(recipient, protocol, message)
+    }
+
+    fn send_to_many(
+        &mut self,
+        _recipients: impl Iterator<Item = PeerId>,
+        _message: StateSyncMessage,
+    ) -> Result<(), NetworkError> {
+        unimplemented!()
```

Comment:
Can we avoid all of these `unimplemented!()` blocks by giving the trait a default implementation of `unimplemented!()` and then having each `impl X for Y` block override the methods it wants to provide? It might clean up a lot of these code blocks.

e.g., see this playground example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b7724fc5e820973160f3b09a8f7a9c95

Reply:
I did, and the compiler complained at me because of something, but I'll go back and do it again.

I believe it's a problem with `async_trait` adding an implicit lifetime, that I couldn't add and the compiler wouldn't accept.  (Kind of a reason async_traits aren't in the original language)

Reply:
So, the compromise is I can do it for everything but the `async fn` because it complains about a lifetime that isn't there, and then when I add it it complains it doesn't exist...